test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Audit/AuditTests.cs

```diff
@@ -389,6 +392,59 @@ private async Task ExecuteAndValidate<T>(Func<Task<FhirResponse<T>>> action, str
                 ae => ValidateExecutedAuditEntry(ae, expectedAction, expectedResourceType, expectedUri, expectedStatusCode, correlationId, expectedAppId, ExpectedClaimKey));
         }
 
+        private async Task ExecuteAndValidateBatch<T>(Func<Task<FhirResponse<T>>> action, List<(string, string, HttpStatusCode)> expectedList)
+            where T : Resource
+        {
+            if (!_fixture.IsUsingInProcTestServer)
+            {
+                // This test only works with the in-proc server with customized middleware pipeline.
+                return;
+            }
+
+            FhirResponse<T> response = null;
+
+            response = await action();
+
+            string correlationId = response.Headers.GetValues(RequestIdHeaderName).FirstOrDefault();
+
+            Assert.NotNull(correlationId);
+
+            string expectedAppId = TestApplications.ServiceClient.ClientId;
+
+            IReadOnlyList<AuditEntry> auditList = _auditLogger.GetAuditEntriesByCorrelationId(correlationId);
+
+            Assert.Equal(18, auditList.Count);
```

Comment:
I think the test should be as simple as possible and doesn't have too many conditions. 1. because more condition in test = harder to read/maintain, 2. more logic means it could have error in the logic itself. Also, the logic in this method is very much tied to the test (E.g., where did 18 come from? Why are there even and odd validations?)